File: test_setup.py
import os
from dotenv import load_dotenv
import google.generativeai as genai

def test_setup():
    print("🔍 Testing setup...")
    
    # Load environment variables from .env file
    load_dotenv()

    # Check API key
    api_key = os.getenv("GOOGLE_API_KEY") or os.getenv("GEMINI_API_KEY")
    if not api_key:
        print("⚠️  GOOGLE_API_KEY or GEMINI_API_KEY not found in environment")
        return False

    # Test API call
    try:
        genai.configure(api_key=api_key)
        model = genai.GenerativeModel('models/gemini-2.0-flash')
        response = model.generate_content("Hi")
        print("✅ Setup successful! API is working.")
        print(f"📝 Response: {response.text[:50]}...")
        return True
    except Exception as e:
        print(f"❌ Error: {e}")
        return False

if __name__ == "__main__":
    test_setup()

File: test_setup.ts
import { GoogleGenerativeAI } from "@google/generative-ai";
import { config } from "dotenv";

async function testSetup() {
  console.log("🔍 Testing setup...");
  
  // Load environment variables from .env file
  config();

  const apiKey = process.env.GOOGLE_API_KEY || process.env.GEMINI_API_KEY;
  if (!apiKey) {
    console.log("⚠️  GOOGLE_API_KEY or GEMINI_API_KEY not found in environment");
    return false;
  }

  try {
    const genAI = new GoogleGenerativeAI(apiKey);
    const model = genAI.getGenerativeModel({ model: "gemini-2.0-flash" });
    const result = await model.generateContent("Hi");
    const response = await result.response;
    console.log("✅ Setup successful! API is working.");
    console.log("📝 Response:", response.text().substring(0, 50) + "...");
    return true;
  } catch (error) {
    console.log("❌ Error:", error);
    return false;
  }
}

testSetup();